appcues/src/main/java/com/appcues/data/model/ExperienceStepFormState.kt

```diff
@@ -0,0 +1,167 @@
+package com.appcues.data.model
+
+import androidx.compose.runtime.mutableStateOf
+import com.appcues.data.model.ExperiencePrimitive.OptionSelectPrimitive
+import com.appcues.data.model.ExperiencePrimitive.TextInputPrimitive
+import com.appcues.data.model.ExperienceStepFormItemState.OptionSelectFormItemState
+import com.appcues.data.model.ExperienceStepFormItemState.TextInputFormItemState
+import com.appcues.data.model.styling.ComponentSelectMode.MULTIPLE
+import com.appcues.data.model.styling.ComponentSelectMode.SINGLE
+import java.util.UUID
+
+internal class ExperienceStepFormState {
+    private var _formItems: HashMap<UUID, ExperienceStepFormItemState> = hashMapOf()
+
+    private var itemIndex = 0
+
+    val formItems: Collection<ExperienceStepFormItemState>
+        get() = _formItems.values.sortedBy { it.index }
+
+    // this can be used by buttons needing to know if they are enabled or not based on required input
+    val isFormComplete = mutableStateOf(true)
+
+    fun getValue(primitive: TextInputPrimitive) =
+        getItem(primitive).text
+
+    fun setValue(primitive: TextInputPrimitive, value: String) {
+        val item = getItem(primitive)
+        item.setValue(value)
+        updateFormItem(item)
+    }
+
+    fun getValue(primitive: OptionSelectPrimitive) =
+        getItem(primitive).values
+
+    fun setValue(primitive: OptionSelectPrimitive, value: String) {
+        val item = getItem(primitive)
+        item.setValue(value)
+        updateFormItem(item)
+    }
```

Comment:
`setValue` now take a single item string, not the full set - and either toggles it ON or OFF based on the previous state of that item.  This allows us to preserve the users selection order.